### PR TITLE
Update select_topk to not use deprecated function

### PR DESCRIPTION
**Pull request by ptigwe**
Replacing the deprecated `jax.ops.index_update` with the suggested
alternative of `arr.at[idx].set(val)`. Another alternative which is to
use masking tricks also yields the same effect is as follows:

``` python
idx_axis0 = jnp.arange(prob_tensor.shape[0])
jnp.sum(idx_axis0 == jnp.expand_dims(idx_axis1, -1), 1)
```

### CODE REVIEW

**Comment by cgarciae** [comment]
LGTM! Thanks a lot @ptigwe!
